src/snovault/elasticsearch/indexer.py

```diff
@@ -143,140 +154,302 @@ def get_related_uuids(request, es, updated, renamed):
-            invalidated = list(all_uuids(request.registry, types))
+        if len(invalidated) > SEARCH_MAX:  # Priority cycle already set up
             flush = True
         else:
-            txns = session.query(TransactionRecord).filter(
-                TransactionRecord.xid >= last_xmin,
-            )
-
-            invalidated = set(invalidated)  # not empty if API index request occurred
-            updated = set()
-            renamed = set()
-            max_xid = 0
-            txn_count = 0
-            for txn in txns.all():
-                txn_count += 1
-                max_xid = max(max_xid, txn.xid)
-                if first_txn is None:
-                    first_txn = txn.timestamp
-                else:
-                    first_txn = min(first_txn, txn.timestamp)
-                renamed.update(txn.data.get('renamed', ()))
-                updated.update(txn.data.get('updated', ()))
 
-            if invalidated:        # reindex requested, treat like updated
-                updated |= invalidated
-
-            result['txn_count'] = txn_count
-            if txn_count == 0 and len(invalidated) == 0:
-                state.send_notices()
-                return result
-
-            if is_testing and is_testing_full:
-                full_reindex = False
-                related_set = set(all_uuids(request.registry))
-            else:
-                (related_set, full_reindex) = get_related_uuids(request, es, updated, renamed)
-            if full_reindex:
-                invalidated = related_set
+            flush = False
+            if last_xmin is None:
+                result['types'] = types = request.json.get('types', None)
+                invalidated = list(all_uuids(request.registry, types))
                 flush = True
             else:
-                invalidated = related_set | updated
-                result.update(
-                    max_xid=max_xid,
-                    renamed=renamed,
-                    updated=updated,
-                    referencing=len(related_set),
-                    invalidated=len(invalidated),
-                    txn_count=txn_count
+                txns = session.query(TransactionRecord).filter(
+                    TransactionRecord.xid >= last_xmin,
                 )
-                if first_txn is not None:
-                    result['first_txn_timestamp'] = first_txn.isoformat()
 
-            if invalidated and not dry_run:
-                # Exporting a snapshot mints a new xid, so only do so when required.
-                # Not yet possible to export a snapshot on a standby server:
-                # http://www.postgresql.org/message-id/CAHGQGwEtJCeHUB6KzaiJ6ndvx6EFsidTGnuLwJ1itwVH0EJTOA@mail.gmail.com
-                if snapshot_id is None and not recovery:
-                    snapshot_id = connection.execute('SELECT pg_export_snapshot();').scalar()
+                invalidated = set(invalidated)  # not empty if API index request occurred
```

Comment:
oh I see, this isn't new code it's just rearranged a little, you moved it into this function